client/timeline/producer/producer.go

```diff
@@ -105,15 +112,23 @@ func (c *Producer) InsertMessages(ctx context.Context, msgs []timeline.Message)
 		err = stream.Send(builder)
 		//TODO if err is invalid/expired sessionID do a handshake automatically
 		if err != nil {
+			metricTopicMessagesErrors.With(prometheus.Labels{"operation":"insert", "topic":c.GetTopic()}).Inc()
 			return errors.Wrap(err, "failed stream")
 		}
+		metricTopicMessagesCounter.With(prometheus.Labels{"operation":"insert", "topic":c.GetTopic()}).Inc()
```

Comment:
the insert operation is succesfully after the stream closes, I would move the code at the end of method and increase it with the number of inserted messages.